src/bin/inx-chronicle/stardust_inx/mod.rs

```diff
@@ -75,12 +72,12 @@ impl InxWorker {
     pub async fn run(&mut self) -> Result<(), InxError> {
         let (start_index, mut inx) = self.init().await?;
 
-        let mut stream = LedgerUpdateStream::new(inx.listen_to_ledger_updates((start_index.0..).into()).await?);
+        let mut stream = inx.listen_to_ledger_updates((start_index.0..).into()).await?;
 
         debug!("Started listening to ledger updates via INX.");
 
         while let Some(ledger_update) = stream.try_next().await? {
-            handle_ledger_update(ledger_update, &mut inx, &self.db).await?;
+            handle_ledger_update(&mut inx, &self.db, ledger_update, &mut stream).await?;
```

Comment:
at this point we can just move the `while let` into `handle_ledger_update` IMO

Reply:
Well the reason I did it this way is to keep the span around the function.

Reply:
oh I see ... damn it!